core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/domain/variable/CustomShadowVariable.java

```diff
@@ -13,7 +13,10 @@
  * Specifies that a bean property (or a field) is a custom shadow of 1 or more {@link PlanningVariable}'s.
  * <p>
  * It is specified on a getter of a java bean property (or a field) of a {@link PlanningEntity} class.
+ *
+ * @deprecated This annotation is deprecated in favor of {@link ShadowVariable} and {@link PiggybackShadowVariable}.
```

Comment:
Typically, the deprecated message is also added at the start of the javadoc, so it appears both in the javadoc popup at the top, as well as in the bottom. You can't tell users enough with what they should replace it :)

Reply:
```suggestion
 * @deprecated Deprecated in favor of {@link ShadowVariable} (nomal) and {@link PiggybackShadowVariable} (if @link variableListenerRef} is used).
```